src/Configuration/Programmer/Programmer_helper.cpp

```diff
@@ -72,6 +72,36 @@ int ConvertIntegerStringToInt(const std::string& str, std::size_t* pos,
   }
 }
 
+int64_t ConvertIntegerStringToInt64(const std::string& str, std::size_t* pos,
```

Comment:
Will need to understand more during our weekly. Might not need this.

Reply:
This `ConvertIntegerStringToInt64` is a wrapper to `stoll` to turn the exception into a warning. During the parsing `extractDeviceList` function, this function is to convert the IdCode and irLen from the extracted string, if it failed, I don't want the exception to happen instead it will return 0 and a warning message.